src/main/java/kostka/moviecatalog/controller/MovieDetailController.java

```diff
@@ -25,10 +35,46 @@ public MovieDetailController(final MovieDetailService movieDetailService) {
      * @param model MVC model.
      * @return name of the html file (detail.html).
      */
-    @GetMapping("/detail/{movieId}")
-    public String getMovieDetail(final @PathVariable Long movieId, final Model model) {
-        MovieDetail movieDetail = movieDetailService.getMovieDetail(movieId);
-        model.addAttribute("movieDetail", movieDetail);
-        return "detail";
+    @GetMapping("/{movieId}")
+    public String getMovieDetail(
+            final @PathVariable Long movieId,
+            final @AuthenticationPrincipal CustomUserDetails user,
+            final Model model) {
+        addMovieDetailModelAttributes(movieId, user.getUserId(), model, "");
+        return MOVIE_DETAIL_VIEW;
+    }
+
+    @GetMapping("/buy/{movieId}")
```

Comment:
I would use another verb. Maybe POST?

Reply:
First I thought nevermind, it is not REST API. But with Thymeleaf, if you buy a movie and then you refresh page, is it not bought again?

Reply:
done via PostMapping ;)